Windows/MainWindow.xaml.cs

```diff
@@ -202,6 +202,40 @@ public partial class MainWindow : Window {
             recentMaps.AddRecentlyOpened((string)mapEditor.GetMapValue("_songName"), newMapFolder);
             recentMaps.Write();
         }
+        internal void InitImportMap(string importMapFolder) {
+            if (mapIsLoaded) {
+                mapEditor.ClearSelectedDifficulty();
+                ClearCoverImage();
+                UnloadSong();
+            }
+
+            // select simfile to import
+            string file = null;
+            var d = new Microsoft.Win32.OpenFileDialog() { Filter = "StepMania simfile|*.sm;*.ssc" };
+            d.Title = "Select a simfile to import";
+            if (d.ShowDialog() == true) {
+                file = d.FileName;
+            } else {
+                return;
+            }
+
+            RagnarockMap beatmap = new RagnarockMap(importMapFolder, true);
```

Comment:
You've moved the beatmap into mapEditor, but here I need direct access to it to perform the conversion. I don't think it's a big deal, as this beatmap is only used for data population during conversion process and then is saved and abandoned for the garbage collector.

Reply:
all good. I moved the beatmap into mapeditor to disentangle some of the classes but in this instance you're directly writing to the files without going through the editor, so it's fine.